Documentation/API.md

```diff
@@ -108,10 +125,23 @@ want to override the default generated behavior for any reason:
 Proto enums are translated to Swift enums in a fairly straightforward manner.
 The resulting Swift enums conform to the `SwiftProtobuf.Enum` protocol which extends
 `RawRepresentable` with a `RawValue` of `Int`.
+The name of the Swift enum is copied directly from the name in the proto file.
+If that name would cause problems for the generated code, the word `Enum` will
+be appended to the name.
+
+Proto2 enums simply have a case for each enum value.
```

Comment:
This seems partial, did you want to talk about proto3 syntax files having the unknown values vs. pushing things into the unknown fields?